actix-server/src/worker.rs

```diff
@@ -289,65 +293,83 @@ impl ServerWorker {
             Arbiter::new()
         };
 
-        #[cfg(not(all(target_os = "linux", feature = "io-uring")))]
-        let arbiter = Arbiter::with_tokio_rt(move || {
-            tokio::runtime::Builder::new_current_thread()
-                .enable_all()
-                .max_blocking_threads(config.max_blocking_threads)
-                .build()
-                .unwrap()
-        });
+        // get actix system context if it is set
+        let sys = System::try_current();
 
-        arbiter.spawn(async move {
-            let fut = factories
-                .iter()
-                .enumerate()
-                .map(|(idx, factory)| {
-                    let fut = factory.create();
-                    async move { fut.await.map(|(t, s)| (idx, t, s)) }
-                })
-                .collect::<Vec<_>>();
-
-            // a second spawn to run !Send future tasks.
-            spawn(async move {
-                let res = join_all(fut)
-                    .await
-                    .into_iter()
-                    .collect::<Result<Vec<_>, _>>();
-                let services = match res {
-                    Ok(res) => res
-                        .into_iter()
-                        .fold(Vec::new(), |mut services, (factory, token, service)| {
-                            assert_eq!(token, services.len());
-                            services.push(WorkerService {
-                                factory,
-                                service,
-                                status: WorkerServiceStatus::Unavailable,
-                            });
-                            services
-                        })
-                        .into_boxed_slice(),
-                    Err(e) => {
-                        error!("Can not start worker: {:?}", e);
-                        Arbiter::current().stop();
-                        return;
-                    }
-                };
+        // TODO: wait for server startup with sync channel
 
-                // a third spawn to make sure ServerWorker runs as non boxed future.
-                spawn(ServerWorker {
-                    rx,
-                    rx2,
-                    services,
-                    counter: WorkerCounter::new(idx, waker_queue, counter_clone),
-                    factories: factories.into_boxed_slice(),
-                    state: Default::default(),
-                    shutdown_timeout: config.shutdown_timeout,
-                });
-            });
-        });
+        std::thread::Builder::new()
+            .name("eofibef".to_owned())
+            .spawn(move || {
+                // forward existing actix system context
+                if let Some(sys) = sys {
+                    System::set_current(sys);
+                }
+
+                let rt = tokio::runtime::Builder::new_current_thread()
```

Comment:
We can pass an Option<tokio::runtime::Handle> to worker and block_on it if possible.